scripts/ci/make/tasks

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+######################################################
+###   PROJECT SPECIFIC FUNCTIONS - RUNTIME ORDER   ###
+######################################################
+
+build() {
+    # dont use the docker build cache if building from a tag
+    print_colorized WARN "building image ${TEST_IMAGE_NAME}"
+    if [[ -n "${GIT_TAG}" ]]; then
+        local build_cache="--no-cache"
+    fi
+    docker build --target anchore-engine-builder ${build_cache} -t "${GIT_REPO}:builder" -f ./Dockerfile .
+    docker build --build-arg ANCHORE_COMMIT="${COMMIT_SHA}" ${build_cache} --build-arg CLI_COMMIT="${CLI_COMMIT_SHA}" -t "${TEST_IMAGE_NAME}" -f ./Dockerfile .
+    print_colorized WARN "successfully built image -- ${TEST_IMAGE_NAME}"
+}
+
+lint() {
+    install_testing_deps pylint
+    print_colorized WARN "linting project"
+    pylint anchore_engine
+    pylint anchore_manager
+}
+
+unit_tests() {
+    install_testing_deps tox
+    print_colorized WARN "running unit tests"
+    tox test/unit
+}
+
+integration_tests() {
+    install_testing_deps tox
+    print_colorized WARN "running integration tests"
+    if [[ "${CI}" == true ]]; then
+        tox test/integration
+    else
+        ./scripts/tests/test_with_deps.sh test/integration/
+    fi
+}
+
+functional_tests() {
+    install_testing_deps anchorecli docker-compose tox
+    print_colorized WARN "running functaional tests"
+    anchore-cli --u admin --p foobar --url http://localhost:8228/v1 system wait --feedsready ''
+    anchore-cli --u admin --p foobar --url http://localhost:8228/v1 system status
+    tox test/functional
+}
+
+push_dev_image() {
+    local dev_image="${DEV_IMAGE_REPO}:${COMMIT_SHA}"
+    local latest_image="${DEV_IMAGE_REPO}:latest"
+    local branch_image="${DEV_IMAGE_REPO}:${GIT_BRANCH}"
+
+    print_colorized WARN "tagging and pushing ${dev_image}"
+    docker tag "${TEST_IMAGE_NAME}" "${dev_image}"
+    docker push "${dev_image}"
+
+    if [[ "${GIT_BRANCH}" == 'master' ]]; then
+        print_colorized WARN "tagging & pushing image -- ${latest_image}"
+        docker tag "${dev_image}" "${latest_image}"
+        docker push "${latest_image}"
+    elif [[ "${GIT_BRANCH}" =~ "${RELEASE_BRANCHES}" ]]; then
+        print_colorized WARN "tagging & pushing image -- ${branch_image}"
+        docker tag "${dev_image}" "${branch_image}"
+        docker push "${branch_image}"
+    fi
+}
+
+e2e_tests() {
+    # Use the RC image if a semver tag is pushed
+    if [[ "${GIT_TAG}" =~ ^v[0-9]+(\.[0-9]+)*$ ]]; then
```

Comment:
this is another instance where logging would be necessary: it relies on a regular expression that will fallback to something else if it doesn't match. Nothing here would advertise this situation. The regex looks fine, but it is obscuring the `else` condition.

Reply:
logging added